Client/src/client.cpp

```diff
@@ -65,3 +67,35 @@ std::vector<std::string> ChronoLogClient::ShowChronicles(std::string &client_id)
 std::vector<std::string> ChronoLogClient::ShowStories(std::string &client_id, const std::string &chronicle_name) {
     return rpcClient_->ShowStories(client_id, chronicle_name);
 }
+uint64_t ChronoLogClient::GetTS()
+{
+	return rpcClient_->GetTS(clientid);
+}
+
+void ChronoLogClient::ComputeClockOffset()
+{
+	uint64_t t1 = CSManager->LocalGetTS(clientid);
+	uint64_t vt1 = GetTS();
+
+	uint64_t vt2 = GetTS();
+	uint64_t t2 = CSManager->LocalGetTS(clientid);
+
+	int64_t offset1 = vt1-t1;
+        int64_t offset2 = vt2-t2;
+
+	double diff = (double)(offset1+offset2)/(double)2; 
```

Comment:
the offset here is the half of the roundtrip network delay , correct?

Reply:
I think we need to minimize the possibility of unequal latency between capturing t1 and vt1 vs capturing t2 and vt2. You may consider delay the capture of local ts as late as possible. For example, you may want to do it inside GetTS(). Same thing for t2 and vt2. You can capture local ts inside GetTS().